lib/Thrift/Transports/Client/TBufferedClientTransport.cs

```diff
@@ -87,7 +94,11 @@ public override void Close()
                 _inputBuffer.Capacity = _bufSize;
             }
 
+#if NETSTANDARD2_1
+            var got = await _inputBuffer.ReadAsync(new Memory<byte>(buffer, offset, length), cancellationToken).ConfigureAwait(false);
```

Comment:
Instead of allocating a new `Memory<byte>`, you could use `MemoryPool<byte>.Shared.Rent()` instead. Check out it's suggested usage [here](https://docs.microsoft.com/en-us/dotnet/standard/memory-and-spans/memory-t-usage-guidelines#memoryt-and-the-ownerconsumer-model).

Reply:
Sorry about all the changes! I thought it would be quick and easy... a couple of tweaks here and there... and it just kept unraveling more and more on me. Taking a look at MemoryPool stuff. I'll reply back.

Reply:
@MikeGoldsmith I tried to make it work with `ReadOnlyMemory<byte>` being passed in on writes and `Memory<byte>` being passed in on reads but it started to get really nasty. Stream doesn't have...
```csharp
ValueTask<int> ReadAsync(Memory<byte> destination, CancellationToken cancellationToken = default)
ValueTask WriteAsync(ReadOnlyMemory<byte> source, CancellationToken cancellationToken = default);
```
...until .NET Standard 2.1 which means to support net46 & standard 2.0 we have to end up with ([], offset, count) pattern in order to write to the UDP socket, which doesn't seem possible from the new memory types?